src/analyzing/Visit_pattern_ids.ml

```diff
@@ -0,0 +1,51 @@
+(* Brandon Wu
+ *
+ * Copyright (C) 2023 r2c
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file license.txt.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * license.txt for more details.
+ *)
+
+module G = AST_generic
+
+class ['self] pat_id_visitor =
+  object (_self : 'self)
+    inherit ['self] AST_generic.iter_no_id_info as super
+
+    method! visit_pattern store pat =
+      match pat with
+      | PatAs (_, (id, id_info))
+      | PatId (id, id_info) ->
+          super#visit_pattern store pat;
+          Common.push (id, id_info) store
+      | PatLiteral _
+      | PatConstructor _
+      | PatRecord _
```

Comment:
Weren't you going to treat records specially to avoid adding variables to the context that should not be in it?

Reply:
This shouldn't add the labels of the variables though, right? We only visit the inner patterns recursively, we don't add the labels (which are `dotted_ident`s) to the env. Correct me if I'm wrong.

Reply:
Actually, it's the other way around here I guess, where we aren't adding the record patterns (not the labels) properly.